### PR TITLE
Minimize docker image

### PR DESCRIPTION
**Pull request by nhynes**
Using an intermediate image reduces the build size from 81 MB to 70 MB. Not bad, I guess.

### CODE REVIEW

**Comment by mitar** [comment]
Interesting. So Rust does not need any dynamic libraries? Binary just works?

One minor point is that timezone is not set in the final image, but probably it does not matter?

File: Dockerfile [hunk 1]
```diff
@@ -22,6 +19,13 @@ RUN \
  apt autoremove --yes && \
  apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
+FROM ubuntu:bionic as oasis-chain-runner
+
+COPY --from=oasis-chain-builder /usr/local/bin/oasis-chain /usr/local/bin
```
Comment (mitar) [inline Dockerfile:24]: Probably copy from `/usr/local/lib/cargo/bin/oasis-chain`? You are copying symlink then?

File: Dockerfile [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
  apt-get update -q -q && \
  apt-get install --yes tzdata curl ca-certificates git build-essential && \
  echo 'UTC' > /etc/timezone && \
  rm /etc/localtime && \
  dpkg-reconfigure tzdata && \
  apt-get upgrade --yes && \
- adduser --system --group oasis --home /nonexistent && \
  curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --no-modify-path --default-toolchain nightly && \
  RUSTFLAGS='-C target-feature=+aes,+ssse3' /usr/local/lib/cargo/bin/cargo install --locked --git https://github.com/oasislabs/oasis-chain.git oasis-chain && \
  cp /usr/local/lib/cargo/bin/oasis-chain /usr/local/bin/oasis-chain && \
```
Comment (mitar) [inline Dockerfile:16]: All lines from here to the end here you could just remove, no? To speedup building. We do not need cleanup anymore then now.

**Comment by nhynes** [comment]
> Interesting. So Rust does not need any dynamic libraries?

yeah. The default `crate-type` is `staticlib` for bin targets (`rlib` for lib targets).

File: Dockerfile (hunk 1 above)
Comment (nhynes) [reply Dockerfile:24]: it's not a symlink. it's `cp`'d

File: Dockerfile (hunk 2 above)
Comment (nhynes) [reply Dockerfile:16]: ehh. it keeps the intermediate layers small 🤷‍♂️

File: Dockerfile (hunk 1 above)
Comment (mitar) [reply Dockerfile:24]: You are right. I do not know why I thought that `/usr/local/bin/oasis-chain` is a symlink.

File: Dockerfile (hunk 2 above)
Comment (mitar) [reply Dockerfile:16]: But one does not pull those layers when pulling the final image?

**Pull request merged**